test/test_jit.py

```diff
@@ -12781,6 +12781,25 @@ def forward(self, z):
 
         self.checkModule(M(), (torch.randn(2, 2),))
 
+    def test_class_compile(self):
+        class B(object):
```

Comment:
this should have a test where the methods call a function defined in the same scope as where the class is defined but in a different scope from the thing that demands the class. I am pretty sure the current code would fail this test.